Providers/Scripts/3.x/Scripts/nxFileInventory.py

```diff
@@ -245,8 +245,23 @@ def GetFileInfo(fname, Links, MaxContentsReturnable, Checksum):
     d['ModifiedDate'] = int(stat_info.st_mtime)
     d['CreatedDate'] = int(stat_info.st_ctime)
     d['FileSize'] = stat_info.st_size
+    # if file size is 0 
+    # dont attempt to read the file
+    if stat_info.st_size == 0:
+       d['Contents'] = ''
+       if Checksum == 'md5' or Checksum == 'sha-256':
+        d['Checksum'] = 0 
+       elif Checksum == "ctime":
+        d['Checksum']= str(int(stat_info.st_ctime))
+       else : # Checksum == "mtime":
+        d['Checksum']= str(int(stat_info.st_mtime))
+       return d
+
     if Checksum == 'md5' or Checksum == 'sha-256':
+       try:
         d['Checksum'] = GetChecksum(fname,Checksum)
+       except:
+        d['Checksum'] = 0 
```

Comment:
Can we make a note somewhere that `d['Checksum'] = 0`  could indicate a file size of 0 OR an error in `GetChecksum`?